bitnami/contour/Chart.yaml

```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: contour
 description: Contour Ingress controller for Kubernetes
-version: 1.2.0
+version: 1.3.0
```

Comment:
You should be bumping the major version, right? I mean, if I enable a feature included in the new CRDs during a helm upgrade it will fail complaining that the feature doesn't exist in the previously installed CRDs.

Reply:
Oh yes I wanted to discuss about that. I've no problem with that. But I think this is not a breaking change. The chart itself does not apply resources which belongs to the CRDs. The CRDs are only used by other applications to have more features than the generic and built-in Ingress-v1 API.

The guys behind projectcontour.io only extend their CRDs but do not remove fields since they went GA with 1.0.0 in november 2019.

Shall I bump it?
Please also read through  this comment here: https://github.com/bitnami/charts/pull/3381#issuecomment-678777617. Eg. when upgrading from chart version `1.1.2` to `1.1.3` (https://github.com/bitnami/charts/commit/04732989c72fb1a03aaed8798d0ae24345190475#diff-b04dc01907aa7a4911696009fecec82f), the upgrade will fail.

Reply:
> Oh yes I wanted to discuss about that. I've no problem with that. But I think this is not a breaking change. The chart itself does not apply resources which belongs to the CRDs. The CRDs are only used by other applications to have more features than the generic and built-in Ingress-v1 API.

Oh... You're right.. In that case the `helm upgrade` won't fail.

> Please also read through this comment here: #3381 (comment). Eg. when upgrading from chart version 1.1.2 to 1.1.3 (0473298#diff-b04dc01907aa7a4911696009fecec82f), the upgrade will fail.

Please correct me if I'm wrong but this job should be run just once, shouldn't it? I mean, why should I re-generate the certificates during a helm upgrade? In that case, I guess we need to use helm hooks to ensure the job is only executed during the installation (skipping upgrades). WDYT?

Reply:
The job generates certifcates which are valid for 365days. I am not sure if `envoy` and `contour` hot-reloads the mounted certificates at runtime when the job finishes (and thus re-generation is done).

I propose that we should also use the `latest` image in the JobSpec to ensure chart upgrade won't fail unless we know exactly how to rotate the certificates "on demand".

Reply:
> I propose that we should also use the latest image in the JobSpec to ensure chart upgrade won't fail unless we know exactly how to rotate the certificates "on demand".

Then, we need to create a specific image for the job. I mean in the values.yaml, we should have sth like:

```yaml
job: 
  image:
    registry: docker.io
    repository: bitnami/contour
    tag: latest
```

The thing is.. We are currently reusing the image taken from:

```yaml
contour:
  image:
    registry: docker.io
    repository: bitnami/contour
    tag: 1.7.0-debian-10-r0
```

That image (used for the main Contour container) **must use immutable tags**, the reasons can be found on this link:
- https://docs.bitnami.com/tutorials/understand-rolling-tags-containers/

> I am not sure if envoy and contour hot-reloads the mounted certificates at runtime when the job finishes (and thus re-generation is done)

I really doubt that it regenerates the certificates. That said, you could implement a way to "force" re-generating them.

IMHO using hooks is the right way to ensure a certain object is created just once.

Reply:
Mh I am unsure what to implement for now. When we bump the major version to `2.0.0` I would introduce the new structure for the job image just to be flexible in the future. If we do not bump the major version we should not use a new structure and fix it with a hook.

We can also say that this issue is not part of this PR and we simply do not bump the major version.

Reply:
Hi guys!

Regarding the bumping of the version, as we are including `bitnami/common` it should be a major one. I mean, when adding these changes in other charts we are bumping it to a major.

I think we could use this major bump to ensure "regenerates the certificates" works fine after upgrading the chart.

Reply:
I studied the hooks documentation here https://helm.sh/docs/topics/charts_hooks/.
Thinking about these hooks we have the following problems:
* `pre-install` seems to early for me, configmap etc. are not yet there
* `post-install` otherwise seems to late, because when we use it with the option `--wait` which most CI/CD systems do, we will run into a timeout because envoy and contour cannot start without an existing secret containing TLS secrets.

Maybe we should involve your VMware colleagues from projectcontour for a discussion?

Reply:
let's involve them to hear their opinion on what would be the best solution.

IMHO, if there is no problem updating certs each time we upgrade, I would go for variant 1 using the new structure.

Reply:
By this time I had a working setup with hooks, I would prefer this solution now because we really use the same image for the certificate generation process.
Let's assume we have a running contour installation in a larger company, which cannot easily upgrade the ingress controller for reasons. But then the certs needs to be regenerated. Who knows if we can generate new certificates with eg. contour-1.8 for a running contour-1.6 installation? :)

Reply:
Ok lgtm!

Reply:
Regarding the version, anyway, we should bump it into a major change, please.

Reply:
Sure, then I will bump it to 2.0.0 and merge the variant2 in.